src/common/browser-adapters/browser-adapter-event-manager.ts

```diff
@@ -51,78 +50,90 @@ export class BrowserAdapterEventManager {
     };
 
     private processDeferredEvents(): void {
-        this.deferredEvents = this.deferredEvents.filter(
-            event => this.processEvent(event.eventType, event.eventArgs, true) === false,
-        );
+        const stillDeferred: DeferredEventDetails[] = [];
+        this.deferredEvents.forEach(deferredEvent => {
```

Comment:
Prefer using for-of loops rather than `Array.forEach` - the former gives better stack traces from `Error`s and is more straightforward to step through in a debugger